test/database.test.js

```diff
@@ -0,0 +1,62 @@
+import {
+  afterAll,
+  beforeAll, beforeEach, describe, expect, it,
+} from 'vitest';
+import { client } from '../connections.js';
+import {
+  addNewSoldier, lookForSoldier, lookForAllSoldiers, dbName, dbCollection,
+} from '../database.js';
+
+beforeAll(async () => {
+  await client.connect();
+});
+
+afterAll(async () => {
+  await client.close();
+});
+
+beforeEach(async () => {
+  await client.db(dbName).collection(dbCollection).deleteMany({});
+});
+
+describe('addNewSoldier function', () => {
+  it('Should insert soldier', async () => {
+    const res = await addNewSoldier({ name: 'Amit' });
+    expect(res.acknowledged).eq(true);
+  });
+});
+
+const testId = {
+  _id: '9033543',
+};
+
+const TestSoldier1 = {
+  _id: '12345',
+  name: 'Hili',
+  rank: '1',
+  limitations: ['none', 'none'],
+};
+
+const TestSoldier2 = {
+  _id: '45678',
+  name: 'Hili',
+  rank: '1',
+  limitations: ['none', 'none'],
+};
+
+describe('lookForSoldier function', () => {
+  it('should find correct soldier', async () => {
+    await addNewSoldier(testId);
+    const res = await lookForSoldier(testId);
+    expect(res._id).eq('9033543');
+  });
+});
+
+describe('lookForAllSoldiers function', () => {
+  it('should find all soldiers', async () => {
```

Comment:
when?
How do i know that `TestSoldier1` and `TestSoldier2` have the same name?

1. You can add it to the test describe (`when ...`)
2. You can add it to your code (how?)

Reply:
I did not mean for them to necessarily have the same name. just wanted to have more than one soldier to make sure that my function finds ALL relevant soldiers and not just one. they have the exacts same properties except the id because I was lazy and copied

Reply:
`"should find all soldiers"` is `expect(res.length).toBe(2);` because:

  `await lookForAllSoldiers({ name: testSoldier1.name })`, 

so the key is the `name` property.